pos_sale_order_load/models/sale_order.py

```diff
@@ -51,41 +48,44 @@ def _prepare_pos_json_line(self, line):
 
     @api.model
     def create_from_ui(self, orders):
-        sale_id = orders[0]["data"].get("sale_order_id")
+        order = orders[0]["data"]
+        sale_id = order.get("sale_order_id")
         if sale_id:
+            # We hack the reference so the sale order will be not found
+            # and so the order will be imported again
+            order["name"] = sale_id
             self = self.with_context(update_pos_sale_order_id=sale_id)
         return super().create_from_ui(orders)
 
     @api.model_create_multi
     def create(self, vals):
-        pos_sale_order_id = self._context("update_pos_sale_order_id")
+        pos_sale_order_id = self._context.get("update_pos_sale_order_id")
         if pos_sale_order_id:
-            sale = self.browse(pos_sale_order_id)
-            sale.write(vals)
+            if len(vals) == 1:
+                # In case the we update the data from the POS we only update the line
+                sale = self.browse(pos_sale_order_id)
+                sale.order_line.unlink()
+                sale.write({"order_line": vals[0]["order_line"]})
+            else:
+                raise NotImplementedError
             return sale
         return super().create(vals)
 
     def _pos_json(self):
+        uuid = self.pos_reference.replace("Order ", "")
```

Comment:
Why ?  Empty string will cause issues in PoS.  

Why not simply sale_order.name or an uuid ?